File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/MedicinalProductInteraction.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class MedicinalProductInteraction : DomainResource, IResource
{
    public Base.CodeableConcept? Type { get; set; }
    public Base.CodeableConcept? Effect { get; set; }
    public Base.ResourceReference[]? Subject { get; set; }
    public Base.CodeableConcept? Incidence { get; set; }
    public Base.CodeableConcept? Management { get; set; }
    public string? Description { get; set; }
    public MedicinalProductInteraction_Interactant[]? Interactant { get; set; }

    public class MedicinalProductInteraction_Interactant : BackboneElement
    {
        public object? Item    
        {
            get
            {
                if (ItemReference is not null)
                {
                    return ItemReference;
                }
        
                if (ItemCodeableConcept is not null)
                {
                    return ItemCodeableConcept;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    ItemReference = (Base.ResourceReference)value;
                    return;
                }
        
                if (value?.GetType() == typeof(Base.CodeableConcept))
                {
                    ItemCodeableConcept = (Base.CodeableConcept)value;
                    return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.ResourceReference? ItemReference { get; set; }
        public Base.CodeableConcept? ItemCodeableConcept { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/MedicinalProductInteraction.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import BackboneElement

@dataclass(kw_only=True)
class MedicinalProductInteractionInteractant(BackboneElement):
    item_reference: Optional[Reference] = None
    item_codeable_concept: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class MedicinalProductInteraction(DomainResource):
    type: Optional[CodeableConcept] = None
    effect: Optional[CodeableConcept] = None
    subject: Optional[List[Reference]] = field(default_factory=list)
    incidence: Optional[CodeableConcept] = None
    management: Optional[CodeableConcept] = None
    description: Optional[str] = None
    interactant: Optional[List[MedicinalProductInteractionInteractant]] = field(default_factory=list)